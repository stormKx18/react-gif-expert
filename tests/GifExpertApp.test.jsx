import { render, screen, fireEvent } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp";


describe('Tests on <GifExpertApp />', () => {
    
    test('Component should match snapshot ', () => {
        const {container} = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    })

    test('Component should include the name of the application', () => {
        render(<GifExpertApp />)
        expect(screen.getByText('GifExpertApp')).toBeTruthy();
    })

    test('Component should show new category when a new category is added', () => {
        render(<GifExpertApp />)

        const inputValue = 'test';
        const inputBox = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(inputBox,{target:{value:inputValue}});
        fireEvent.submit(form);

        expect(screen.getByText(inputValue)).toBeTruthy();
        expect(screen.getAllByRole('heading',{level:3}).length).toBe(2);
    })

    test('Component should not show duplicated categories', () => {
        render(<GifExpertApp />)

        const inputValue = 'test';
        const inputBox = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(inputBox,{target:{value:inputValue}});
        fireEvent.submit(form);

        fireEvent.input(inputBox,{target:{value:inputValue}});
        fireEvent.submit(form);

        expect(screen.getAllByText(inputValue).length).toBe(1);
    })
    
    
    
})
