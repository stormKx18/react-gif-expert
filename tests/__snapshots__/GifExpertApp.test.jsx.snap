// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> Component should match snapshot  1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Loading ...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
